firtool-resolver/src/Main.scala

```diff
@@ -164,22 +165,42 @@ object Resolve {
     val destFile: File = destBin.toFile
 
     // Check if binary already exists
-    // Copying a file is not thread-safe
-    this.synchronized {
+    if (destFile.isFile()) {
+      logger.debug(s"Firtool binary $destFile already exists")
+    } else {
```

Comment:
This else portion of this has gotten pretty chunky. Given that the code has already uses return, consider if doing an early exit here would make things more readable by removing indentation. I.e.:

```suggestion
    if (destFile.isFile()) {
      logger.debug(s"Firtool binary $destFile already exists")
      return Right(FirtoolBinary(destFile, version))
    }
```